modules/material-management/test/cryptographic_material.test.ts

```diff
@@ -97,12 +97,24 @@ describe('decorateCryptographicMaterial', () => {
     expect(() => test.unencryptedDataKeyLength).to.throw()
   })
 
-  it('Precondition: If the unencryptedDataKey has not been set, it should not be settable later.', () => {
+  it('Precondition: the unencryptedDataKey must not be Zeroed out.', () => {
+    const suite = new NodeAlgorithmSuite(AlgorithmSuiteIdentifier.ALG_AES128_GCM_IV12_TAG16)
+    const test = decorateCryptographicMaterial((<any>{ suite, keyringTrace: [] }), KeyringTraceFlag.WRAPPING_KEY_GENERATED_DATA_KEY)
+    const dataKey = new Uint8Array(suite.keyLengthBytes).fill(1)
+    const trace = { keyNamespace: 'k', keyName: 'k', flags: KeyringTraceFlag.WRAPPING_KEY_GENERATED_DATA_KEY }
+    test.setUnencryptedDataKey(dataKey, trace)
+    test.zeroUnencryptedDataKey()
+    expect(() => test.unencryptedDataKeyLength).to.throw('unencryptedDataKey has been zeroed.')
+  })
+
+  it(`Precondition: If the unencryptedDataKey has not been set, it should not be settable later.
+      Precondition: If the udkForVerification has not been set, it should not be settable later.`, () => {
     const suite = new NodeAlgorithmSuite(AlgorithmSuiteIdentifier.ALG_AES128_GCM_IV12_TAG16)
     const test = decorateCryptographicMaterial((<any>{ suite, keyringTrace: [] }), KeyringTraceFlag.WRAPPING_KEY_GENERATED_DATA_KEY)
     test.zeroUnencryptedDataKey()
     const dataKey = new Uint8Array(suite.keyLengthBytes).fill(1)
     const trace = { keyNamespace: 'k', keyName: 'k', flags: KeyringTraceFlag.WRAPPING_KEY_GENERATED_DATA_KEY }
+    // It is very hard to test this perfectly.  However, this tests the spirit.
```

Comment:
What is the "spirit" that is actually being tested here, and what test is missing for the above preconditions?

This looks to me like were testing the postcondition at https://github.com/awslabs/aws-encryption-sdk-javascript/blob/master/modules/material-management/src/cryptographic_material.ts#L341

I'm not familiar with the precise definition of pre/post condition as used in this codebase, but if we aren't checking something and throwing an error on the precondition does it make sense to call it a precondition?

Reply:
If the unencryptedDataKey has not been set, it should not be settable later.

i.e. if you create materials, then `test.zeroUnencryptedDataKey()`, you should not be able to set an unencrypted data key later.

Reply:
Ok, I see the behavior now. This test depends on setUnencryptedDataKey maintaining a behavior where it checks whether the unencryptedDataKey/udkForVerification has been set. This makes the test more brittle than I'd like for the precondition as a whole, but at the very least we ensure with this test that the case really we care about ("you can't set after zeroing) is taken care of.